pkg/storage/utils/localfs/upload.go

```diff
@@ -329,10 +329,24 @@ func (upload *fileUpload) FinishUpload(ctx context.Context) error {
 		}
 	}
 
-	err := os.Rename(upload.binPath, np)
+	//err := os.Rename(upload.binPath, np)
+	//
+	//os.Rename is replaced by the code below to allow for the DataDirectory
+	//and Uploads directory to be on diferent file systems.
+	inputFile, err := os.Open(upload.binPath)
 	if err != nil {
 		return err
 	}
+	outputFile, err := os.Create(np)
+	if err != nil {
+		return err
+	}
+	_, err = io.Copy(outputFile, inputFile)
+	if err != nil {
+		return err
+	}
+	inputFile.Close()
+	outputFile.Close()
```

Comment:
Can we `defer` the `.Close()` calls? Otherwise if `os.Create(np)` fails the `inputFile` will not be closed